src/Symfony/Component/Notifier/Bridge/Lox24/Tests/Lox24TransportTest.php

```diff
@@ -294,28 +289,56 @@ public function testResponseStatusCodeNotEqual201()
             ],
         );
 
-        $transport = new Lox24Transport('user', 'token', 'testFrom', [], $this->client);
+        $transport = new Lox24Transport('user', 'token', 'testFrom', [], $client);
 
         $message = new SmsMessage('+1411111111', 'test text');
         $transport->send($message);
     }
 
-    private function assertRequestBody(
+    public function mockHttpClient(
         array $bodyOverride = [],
         array $headersOverride = [],
-        int $responseStatus = 200,
-        array $responseContent = [],
-    ): void {
-        $body = array_merge(self::REQUEST_BODY, $bodyOverride);
+        int   $responseStatus = 200,
+        array $responseContent = []
+    ): MockHttpClient {
+        $body = json_encode(array_merge(self::REQUEST_BODY, $bodyOverride));
         $headers = array_merge(self::REQUEST_HEADERS, $headersOverride);
-        $response = $this->createMock(ResponseInterface::class);
-        $response->expects($this->once())->method('getStatusCode')->willReturn($responseStatus);
-        $response->expects($this->once())->method('toArray')->willReturn($responseContent);
-        $this->client->expects($this->once())
-                     ->method('request')
-                     ->with('POST', 'https://api.lox24.eu/sms', [
-                         'body' => $body,
-                         'headers' => $headers,
-                     ])->willReturn($response);
+
+        $factory = function ($method, $url, $options) use (
+            $body,
+            $headers,
+            $responseStatus,
+            $responseContent
+        ): MockResponse {
+            $this->assertEquals('POST', $method);
+            $this->assertEquals('https://api.lox24.eu/sms', $url);
+            $this->assertHeaders($headers, $options['headers']);
+            $this->assertJsonStringEqualsJsonString($body, $options['body']);
+
+            return new MockResponse(json_encode($responseContent), [
```

Comment:
You can use JsonMockRrsponse to be able to just pass an array instead of json

Reply:
thanks! done.